### PR TITLE
Adding new modifiers [_(noop), removeparam, redirect]

### PR DESCRIPTION
**Pull request by zloyden**

### CODE REVIEW

File: src/compatibility-tables/modifiers/_(noop).yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+adg_os_any:
+  name: _(noop)
+  negatable: false
+  docs: https://adguard.com/kb/general/ad-filtering/create-own-filters/#noop-modifier
+  description: noop modifier does nothing and can be used solely to increase rules' readability. It consists of a sequence of underscore characters (_) of any length and can appear in a rule as many times as needed.
```
Comment (slavaleleka) [inline src/compatibility-tables/modifiers/_(noop).yml:5]:
```suggestion
  description: |-
    The noop modifier does nothing and can be used solely to increase rules' readability.
    It consists of a sequence of underscore characters (_) of any length
    and can appear in a rule as many times as needed.
```

File: src/compatibility-tables/modifiers/_(noop).yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
+adg_os_any:
+  name: _(noop)
```
Comment (scripthunter7) [inline src/compatibility-tables/modifiers/_(noop).yml:2]: Just `_` or `_(noop)`? It is important because here we specify the name of the modifier

Comment (zloyden) [reply src/compatibility-tables/modifiers/_(noop).yml:2]: Only `_` as it mentioned above. I'll change it.

File: src/compatibility-tables/modifiers/_(noop).yml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  negatable: false
+  docs: https://adguard.com/kb/general/ad-filtering/create-own-filters/#noop-modifier
+  description: |-
+    The noop modifier does nothing and can be used solely to increase rules' readability.
```
[the remark this answers is not in the document]
Comment (Alex-302) [reply src/compatibility-tables/modifiers/_(noop).yml:6]: need to add  to platforms  (one or more `_`)

```
  # TODO: implement validation
  # value_format: 
```

like here https://github.com/AdguardTeam/AGLint/blob/master/src/compatibility-tables/modifiers/stealth.yml

File: src/compatibility-tables/modifiers/redirect.yml [hunk 4]
```diff
@@ -0,0 +1,20 @@
+adg_os_any:
```
Comment (Alex-302) [inline src/compatibility-tables/modifiers/redirect.yml:1]: need to add  to platforms 

```
  # TODO: implement validation
  # value_format: 
```

like here https://github.com/AdguardTeam/AGLint/blob/master/src/compatibility-tables/modifiers/stealth.yml

File: src/compatibility-tables/modifiers/removeparam.yml [hunk 5]
```diff
@@ -0,0 +1,20 @@
+adg_os_any:
```
Comment (Alex-302) [inline src/compatibility-tables/modifiers/removeparam.yml:1]: need to add  to platforms 

```
  # TODO: implement validation
  # value_format: 
```

like here https://github.com/AdguardTeam/AGLint/blob/master/src/compatibility-tables/modifiers/stealth.yml

File: src/compatibility-tables/modifiers/redirect.yml [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+  name: redirect
+  assignable: true
```
Comment (scripthunter7) [inline src/compatibility-tables/modifiers/redirect.yml:3]: Please add a TODO comment that the value validation should be done later (it doesn't need to be ready in this PR)

File: src/compatibility-tables/modifiers/removeparam.yml [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+  name: removeparam
+  assignable: true
```
Comment (scripthunter7) [inline src/compatibility-tables/modifiers/removeparam.yml:3]: Please make the TODO comment here as well

**Review by scripthunter7 — CHANGES_REQUESTED** [review]
Please see my comments

File: src/compatibility-tables/modifiers/_(noop).yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (scripthunter7) [reply src/compatibility-tables/modifiers/_(noop).yml]: It is important that in **_all_** cases the **_exact name of the modifier_** should be specified, nothing more

**Review by scripthunter7 — COMMENTED** [review]
Maybe change the name of `_(noop).yml` to `_.yml` or `noop.yml`?

**Comment by Alex-302** [comment]
> noop.yml

I think it will be ok.

File: src/compatibility-tables/modifiers/removeparam.yml (hunk 5 above)
Comment (zloyden) [reply src/compatibility-tables/modifiers/removeparam.yml:1]: Done.

**Comment by zloyden** [comment]
I had read your comments and made changes. Please review it again:)

File: src/compatibility-tables/modifiers/noop.yml [hunk 8]
```diff
@@ -0,0 +1,54 @@
+adg_os_any:
+  name: _
+  negatable: false
+  docs: https://adguard.com/kb/general/ad-filtering/create-own-filters/#noop-modifier
+  description: |-
+    The noop modifier does nothing and can be used solely to increase rules' readability.
+    It consists of a sequence of underscore characters (_) of any length
+    and can appear in a rule as many times as needed.
+  # TODO: implement validation
```
Comment (scripthunter7) [inline src/compatibility-tables/modifiers/noop.yml:9]: By default `assignable` is false, and as i recall, `_` is not assignable. Please move these TODO comments to `redirect` and `removeparam`

**Review by scripthunter7 — CHANGES_REQUESTED** [review]

File: src/compatibility-tables/modifiers/noop.yml (hunk 8 above)
Comment (zloyden) [reply src/compatibility-tables/modifiers/noop.yml]: Yes, it's correct. I moved the comments to those files.

**Review by scripthunter7 — APPROVED** [review]
LGTM, but please wait for @Alex-302 and @slavaleleka too

**Review by Alex-302 — APPROVED** [review]

**Review by slavaleleka — APPROVED** [review]

**Pull request merged**